src/lib/es6.d.ts

```diff
@@ -447,14 +447,17 @@ interface IteratorResult<T> {
 }
 
 interface Iterator<T> {
-    [Symbol.iterator](): Iterator<T>;
     next(): IteratorResult<T>;
+    return?(value?: any): IteratorResult<T>;
+    throw?(e?: any): IteratorResult<T>;
 }
 
 interface Iterable<T> {
     [Symbol.iterator](): Iterator<T>;
 }
 
+interface IterableIterator<T> extends Iterable<T>, Iterator<T> { }

```

Comment:
I need to fix this because it's not deeply iterable, so it won't work if you do:
```ts
var iter: IterableIterator<any>;
iter[Symbol.iterator]()[Symbol.iterator]();
```
The type IterableIterator needs to be recursive for this to work.